packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -2080,6 +2081,150 @@ void main() {
     scrollable.controller.jumpTo(0);
     await verifyVisibility(false, #right, false, #right);
   });
+
+  testWidgets('iOS text selection handle visibility', (WidgetTester tester) async {
+    debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+    final GlobalKey<EditableTextState> editableTextKey =
+        GlobalKey<EditableTextState>();
+
+    const String testText = 'XXXXX          XXXXX';
+    final TextEditingController controller = TextEditingController(text: testText);
+
+    final Widget widget = MaterialApp(
+      home: Align(
+        alignment: Alignment.topLeft,
+        child: Container(
+          child: SizedBox(
+            width: 100,
+            child: EditableText(
+              key: editableTextKey,
+              controller: controller,
+              focusNode: FocusNode(),
+              style: Typography(platform: TargetPlatform.iOS).black.subhead,
+              cursorColor: Colors.blue,
+              backgroundCursorColor: Colors.grey,
+              selectionControls: cupertinoTextSelectionControls,
+              keyboardType: TextInputType.text,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.pumpWidget(widget);
+
+    final EditableTextState state =
+        tester.state<EditableTextState>(find.byType(EditableText));
+    final RenderEditable renderEditable = state.renderEditable;
+    final Scrollable scrollable = tester.widget<Scrollable>(find.byType(Scrollable));
+
+    bool leftVisibleBefore = false;
+    bool rightVisibleBefore = false;
+
+    Future<void> verifyVisibility(
+      bool leftVisible,
```

Comment:
maybe call this expectedLeftVisible etc?

Might be worth checking this in RTL too in which case it's really expectedStartVisible

Reply:
I just added a very simple RTL test.  I tried copying one of these big tests and making it RTL, but I've been fighting with it all day to get it to pass.  I'm seeing lots of weird behavior that I don't see when I manually test the app myself.  I think it's better to just have a simple one than for me to hack in a copy of these that I don't understand.